js/dropdown.js

```diff
@@ -0,0 +1,245 @@
+            this.$input = this.$('input');
+        },
+
+        setUpItems: function() {
+            var $options = this.$('li');
+            $options.each(function(index, el) {
+                new DropDownOption({
+                    parent: this,
+                    el: el
+                });
+            }.bind(this));
+        },
+
+        addEventListeners: function() {
+            this.$button.on({
+                'mousedown touchstart': this.onStartInteraction,
+                click: this.onButtonClick
+            });
+            this.$list.on('blur', this.onListBlur);
+            $(document).on('keydown', this.onKeyDown);
+        },
+
+        onStartInteraction: function() {
+            this.wasOpen = this.isOpen();
+        },
+
+        deselectAll: function() {
+            this.placeholder.deselect();
+            this.options.forEach(function(option) {
+                option.deselect();
+            });
+        },
+
+        onButtonClick: function(event) {
+            if (this.wasOpen || this.isOpen()) {
+                // click toggle list:
+                // if the list is open and the button is clicked
+                // the blur event will fire first closing the list.
+                // this code is to prevent the list from reopening
+                // in that situation
+                return this.wasOpen = false;
+            }
+            this.toggleOpen();
+            var option = this.getFirstSelectedItem() || this.placeholder;
+            if (!option) return;
+            option.reselect();
+            option.scrollTo();
+            this.$list.focus();
+        },
+
+        isOpen: function() {
+            return !this.$list.hasClass('hidden');
+        },
+
+        toggleOpen: function(open) {
+            if (open === undefined) open = !this.isOpen();
+            this.$button.attr('aria-expanded', open ? 'true' : 'false');
+            var name = open ? 'openList' : 'closeList';
+            this.settings[name].call(this, this);
+            this.trigger(name, this);
+        },
+
+        onListBlur: function() {
+            this.toggleOpen(false);
+            this.removeActiveDescendantId();
+        },
+
+        onKeyDown: function(event) {
+            if (!this.isOpen()) return;
+            var option = this.getFirstSelectedItem() || this.placeholder;
+            switch (event.keyCode) {
+                case 38: // UP
+                    event.preventDefault();
+                    option = option.getPrevious() || option.getLast();
+                    break;
+                case 40: // DOWN
+                    event.preventDefault();
+                    option = option.getNext() || option.getFirst();
+                    break;
+                case 13: // ENTER
+                case 32: // SPACE
+                    event.preventDefault();
+                    this.$button.focus();
+                    return;
+                case 27: // ESCAPE
```

Comment:
This could just be added to the cases above (13, 32)

Reply:
Indeed it can. Ta 👍

Reply:
Yeah, looks good